### PR TITLE
feat: pkg-util-github-pullrequest enhancement

### PR DESCRIPTION
**Pull request by jxs1211**
Signed-off-by: xian-jie.shen <327411586@qq.com>

## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests

## Description
add ut for pullrequest

## Related Issues
see https://github.com/devstream-io/devstream/issues/833 for the details

## New Behavior (screenshots if needed)
image

### CODE REVIEW

**Comment by jxs1211** [comment]
@IronCore864 I try to make a ut by using `ginkgo` this time.

**Review by iyear — APPROVED** [review]
Nice testing!

File: pkg/util/github/github_test.go [hunk 1]
```diff
@@ -60,51 +59,64 @@ func testBody(t *testing.T, r *http.Request, want string) {
 	}
 }
 
-// setup sets up a test HTTP server along with a github.Client that is
+// Setup sets up a test HTTP server along with a github.Client that is
```
Comment (hxcGit) [inline pkg/util/github/github_test.go:62]: Some functions are the same. Is it necessary to expose them?

File: pkg/util/github/pullrequest_test.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
+package github_test
```
Comment (iyear) [inline pkg/util/github/pullrequest_test.go:1]: may be the pkg name is `github`

Comment (hxcGit) [reply pkg/util/github/pullrequest_test.go:1]: I think it's OK. 
`Test files that declare a package with the suffix "_test" will be compiled as a
separate package, and then linked and run with the main test binary.`

Refer to https://pkg.go.dev/cmd/go/internal/test

Comment (iyear) [reply pkg/util/github/pullrequest_test.go:1]: First time I've seen it used in this way. I've looked at several go open source projects and haven't seen this way.
https://github.com/moby/moby/tree/master/pkg/archive
https://github.com/kubernetes/kubernetes/tree/master/pkg/kubelet

I think the `_test` suffix describes the file name but not the package name.

Comment (aFlyBird0) [reply pkg/util/github/pullrequest_test.go:1]: > may be the pkg name is `github`

I agree with you. @steinliber and I usually remove "_test" suffix, or you have to exported every function and vars you want to test.

File: pkg/util/github/github_test.go (hunk 1 above)
Comment (aFlyBird0) [reply pkg/util/github/github_test.go:62]: > Some functions are the same. Is it necessary to expose them?

I agree with you.

**Comment by hxcGit** [comment]
@jxs1211  Hi，if you have time please check whether the above comments need to be adjusted and resolve the conflict.

File: pkg/util/github/pullrequest_test.go (hunk 2 above)
Comment (jxs1211) [reply pkg/util/github/pullrequest_test.go:1]: Go have 2 types test:
1 internal test(package name don't have suffix `_test`): mainly use to test unexported function.
2 external test(package name have suffix `_test`): mainly use to test exported function.

In this case, I prefer to use the external test for this package, and `ginkgo` also generates a test file with a package name suffix '_test'.

**Comment by jxs1211** [comment]
> @jxs1211 Hi，if you have time please check whether the above comments need to be adjusted and resolve the conflict.

@hxcGit I’ve rewritten the test and resolved all the conflicts. The coverage hit 100% now.

**Comment by iyear** [comment]
@jxs1211 Please check the CI error, I will merge it in as soon as possible after solving.

**Pull request merged**